File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        a=1
        b=1
        num_zero = 0
        for i in nums:
            if (i!=0):
             a*=i
             b*=i
            else:
                num_zero +=1
                b*=i
            
        if num_zero >=2 :
            return[0 for _ in range(len(nums))]
        for i in range(len(nums)):
            if (nums[i]==0):
                nums[i] = a
            else: 
                nums[i] = b//nums[i]
        return nums

File: Data Structures & Algorithms/majority-element/submission-1.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        std::map<int, int> m;
        
        // Remplir la map - PAS BESOIN de boucle while!
        for (int i = 0; i < nums.size(); i++) {
            m[nums[i]]++;  // C'est tout!
        }
        
        // Chercher le majority element
        for (auto it = m.begin(); it != m.end(); ++it) {
            if (it->second > (int)nums.size() / 2) {
                return it->first;
            }
        }
        
        return -1;  // Jamais atteint
    }
};